### PR TITLE
add sops-yubikey support

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: pkgs/sops-yubikey/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  sops,
+  fetchFromGitHub,
+  buildGoModule,
+}:
+
+let
+  version = "2024-03-18";
+in
+buildGoModule {
+  name = "sops-yubikey-${version}";
```
Comment (SuperSandro2000) [inline pkgs/sops-yubikey/default.nix:11]:
```suggestion
  pname = "sops-yubikey";
  inherit version;
```

**Comment by NovaViper** [comment]
@Mic92 Hey would it be possible to add this into the Home-manager module aswell? I just got it to work with the NixOS module perfectly but only to realize there is no such support made for the Home-Manager module. I get this error when trying to do run it with Home-Manager so:
```
Jun 11 16:39:19 ryzennova cxjq971my8x38hkzvc79fi35s200xngs-sops-nix-user[1018837]: [AGE]         time="2024-06-11T16:39:19-05:00" level=warning msg="could not read value for age-plugin-yubikey: standard input is not a terminal, and /dev/tty is not available: open /dev/tty: no such device or address"
Jun 11 16:39:19 ryzennova cxjq971my8x38hkzvc79fi35s200xngs-sops-nix-user[1018837]: /nix/store/53d071k93bf74clq4bsj71xg57cys0dq-sops-install-secrets-0.0.1/bin/sops-install-secrets: Failed to decrypt '/nix/store/2pz3j4hym109mljaifnyb9qzv9zzr2br-srv_dev_disk_by_uuid_5aaed6a3_d2c7_4623_b121_5ebb8d37d930_Backups': Error getting data key: 0 successful groups required, got 0
Jun 11 16:39:19 ryzennova systemd[3431]: sops-nix.service: Main process exited, code=exited, status=1/FAILURE
```

And this is my config for sops on Home-Manager
```
{
  config,
  inputs,
  ...
}: {
  imports = [inputs.sops-nix.homeManagerModules.sops];

  sops = {
    age = {
      #plugins = with pkgs; [age-plugin-yubikey];
      sshKeyPaths = [];
      keyFile = "${config.home.homeDirectory}/.config/sops/age/keys.txt";
    };
    gnupg = {
      #home = "${config.home.homeDirectory}/.gnupg";
      sshKeyPaths = [];
    };
  };
}
```

**Comment by OliverGeneser** [comment]
What is the status on this PR? Is any help required?

**Comment by Mic92** [comment]
@OliverGeneser you can help if you want. Rather than my pull request this should than use https://github.com/FiloSottile/age/pull/591 and https://github.com/getsops/sops/pull/1641
And instead of the yubikey plugin I would now actually prefer https://github.com/olastor/age-plugin-fido2-hmac because it not only works with yubikeys but all sorts of fido2 token (less vendor login)

**Comment by brianmcgee** [comment]
> Rather than my pull request this should than use [FiloSottile/age#591](https://github.com/FiloSottile/age/pull/591) and https://github.com/getsops/sops/pull/1641

I've done that here https://github.com/Mic92/sops-nix/pull/680

I recently updated it with https://github.com/getsops/sops/pull/1641#issuecomment-2489611978

**Comment by Mic92** [comment]
super seeded by https://github.com/Mic92/sops-nix/pull/680

**Pull request closed without merging**